ratpack-cassandra-tracing/src/main/java/smartthings/ratpack/cassandra/zipkin/TracedSession.java

```diff
@@ -0,0 +1,31 @@
+package smartthings.ratpack.cassandra.zipkin;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+import com.github.kristofa.brave.Brave;
+import com.google.inject.Inject;
+import smartthings.ratpack.cassandra.AbstractSession;
+
+public final class TracedSession extends AbstractSession {
+
+	private final Brave brave;
+	private final Cluster cluster;
+	private final String keyspace;
+	private final CassandraTracingModule.Config config;
+
+	@Inject
+	public TracedSession(Cluster cluster, Brave brave, CassandraTracingModule.Config config) {
+		super(cluster, config.getKeyspace());
+		this.brave = brave;
+		this.cluster = cluster;
+		this.keyspace = config.getKeyspace();
+		this.config = config;
+	}
+
+	protected Session createDelegate() {
+		Session session = (keyspace != null && !keyspace.equals("")) ? cluster.connect(keyspace) : cluster.connect();
+
+		return smartthings.brave.cassandra.TracedSession.create(session, brave, config.getServiceName());
```

Comment:
We might want to consider using a different trace context.  I can't think of a good reason it needs to interact with the Ratpack lifecycle at that level. Even a thread local context might be suitable for our needs.

I will try to do some investigation work later today.